src/EprPrnIntegration.Common/Service/ServiceBusProvider.cs

```diff
@@ -1,45 +1,126 @@
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("SendFetchedNpwdPrnsToQueue failed to add message on Queue with exception: {exception}", ex);
+                throw;
+            }
+        }
+
+        public async Task SendDeltaSyncExecutionToQueue(DeltaSyncExecution deltaSyncExecution)
+        {
+            try
+            {
+                await using var sender = serviceBusClient.CreateSender(config.Value.DeltaSyncQueueName);
+                using var messageBatch = await sender.CreateMessageBatchAsync();
+
+                var executionMessage = JsonSerializer.Serialize(deltaSyncExecution);
+                var message = new ServiceBusMessage(executionMessage)
+                {
+                    ContentType = "application/json"
+                };
+
+                var existingMessage = await ReceiveDeltaSyncExecutionFromQueue(deltaSyncExecution.SyncType);
+
+                if (existingMessage != null)
+                {
+                    if (!messageBatch.TryAddMessage(message))
+                    {
+                        logger.LogWarning("SendDeltaSyncExecutionToQueue - The updated message for SyncType: {SyncType} is too large to fit in the batch.", deltaSyncExecution.SyncType);
+                    }
+                    logger.LogInformation("SendDeltaSyncExecutionToQueue - Updated existing message for SyncType: {SyncType} in the queue", deltaSyncExecution.SyncType);
+                }
+                else
+                {
+                    if (!messageBatch.TryAddMessage(message))
+                    {
+                        logger.LogWarning("SendDeltaSyncExecutionToQueue - The new message for SyncType: {SyncType} is too large to fit in the batch.", deltaSyncExecution.SyncType);
                     }
+                    logger.LogInformation("SendDeltaSyncExecutionToQueue - Created new message for SyncType: {SyncType} in the queue", deltaSyncExecution.SyncType);
                 }
+
                 await sender.SendMessagesAsync(messageBatch);
-                _logger.LogInformation("SendFetchedNpwdPrnsToQueue - A batch of {MessageBatchCount} messages has been published to the queue: {queue}", messageBatch.Count, _serviceBusConfig.Value.FetchPrnQueueName);
+                logger.LogInformation("SendDeltaSyncExecutionToQueue - A batch of {MessageBatchCount} messages has been published to the queue: {queue}", messageBatch.Count, config.Value.DeltaSyncQueueName);
             }
             catch (Exception ex)
             {
-                _logger.LogError("SendFetchedNpwdPrnsToQueue failed to add message on Queue with exception: {exception}", ex);
+                logger.LogError("SendDeltaSyncExecutionToQueue failed to add message on Queue with exception: {exception}", ex);
                 throw;
             }
         }
+
+        public async Task<DeltaSyncExecution?> ReceiveDeltaSyncExecutionFromQueue(NpwdDeltaSyncType syncType)
```

Comment:
A comment would help readers who don't necessarily know the purpose of "ReceiveDeltaSyncExecutionFromQueue" which is a bit of a mouth full

Reply:
added comments